domain/machine/state/state_test.go

```diff
@@ -140,3 +140,21 @@ func (s *stateSuite) TestGetMachineLifeNotFound(c *gc.C) {
 	_, err := s.state.GetMachineLife(context.Background(), "666")
 	c.Assert(errors.IsNotFound(err), jc.IsTrue)
 }
+
+func (s *stateSuite) TestListAllMachines(c *gc.C) {
+	err := s.state.CreateMachine(context.Background(), "666", "3", "1")
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.CreateMachine(context.Background(), "667", "4", "2")
+	c.Assert(err, jc.ErrorIsNil)
+
+	machines, err := s.state.ListAllMachines(context.Background())
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(machines, gc.DeepEquals, []string{"666", "667"})
```

Comment:
This test could have intermittent failures over its life. There is not guarantee of order returned in sql. You need to sort both slices and compare them.